### PR TITLE
rfb: add RFB frames

### PR DESCRIPTION
**Pull request by hsadia538**
- [X] I have read the contributing guide lines at https://suricata.readthedocs.io/en/latest/devguide/codebase/contributing/contribution-process.html
- [X] I have signed the Open Information Security Foundation contribution agreement at https://suricata.io/about/contribution-agreement/
- [ ] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket: https://redmine.openinfosecfoundation.org/issues/5717
 
Previous PR: #8338

Describe changes:
- Update`frame_len` in the parsing functions
- Fix `commit check` issue

suricata-verify-pr: 1050

### CODE REVIEW

File: rust/src/rfb/rfb.rs [hunk 1]
```diff
@@ -17,16 +17,24 @@
 
 // Author: Frank Honza <frank.honza@dcso.de>
 
+
```
Comment (inashivb) [inline rust/src/rfb/rfb.rs]: nit: extra line?

File: rust/src/rfb/rfb.rs [hunk 2]
```diff
@@ -17,16 +17,24 @@
...
 use std;
 use std::ffi::CString;
-use crate::core::{ALPROTO_UNKNOWN, AppProto, Flow, IPPROTO_TCP};
+use crate::core::{AppProto, Flow, ALPROTO_UNKNOWN, IPPROTO_TCP};
```
Comment (inashivb) [inline rust/src/rfb/rfb.rs]: nit: doesn't look like anything should change here as a part of this commit

File: rust/src/rfb/rfb.rs [hunk 3]
```diff
@@ -151,11 +159,14 @@ impl RFBState {
...
 
-    fn parse_request(&mut self, input: &[u8]) -> AppLayerResult {
+    fn parse_request(&mut self, flow: *const Flow, stream_slice: StreamSlice) -> AppLayerResult {
         // We're not interested in empty requests.
+        let input = stream_slice.as_slice();
+
         if input.is_empty() {
             return AppLayerResult::ok();
         }
+        let _pdu = Frame::new(flow, &stream_slice, input, -1_i64, RFBFrameType::Pdu as u8);
```
Comment (inashivb) [inline rust/src/rfb/rfb.rs:167]: any variables that are being used should not be prefixed with an `_`.

File: rust/src/rfb/rfb.rs [hunk 4]
```diff
@@ -226,7 +245,11 @@ impl RFBState {
                             consumed += current.len() - rem.len();
                             current = rem;
 
-                            self.state = parser::RFBGlobalState::TCSecurityResult;
+                        if let Some(frame) = &_pdu {
+                            frame.set_len(flow, consumed as i64);
+                        }
+
+                        self.state = parser::RFBGlobalState::TCSecurityResult;
```
Comment (inashivb) [inline rust/src/rfb/rfb.rs:256]: nit: improper indent

File: rust/src/rfb/rfb.rs [hunk 5]
```diff
@@ -248,7 +271,11 @@ impl RFBState {
                             consumed += current.len() - rem.len();
                             current = rem;
 
-                            self.state = parser::RFBGlobalState::TCServerInit;
+                        if let Some(frame) = &_pdu {
+                            frame.set_len(flow, consumed as i64);
+                        }
+
+                        self.state = parser::RFBGlobalState::TCServerInit;
```
Comment (inashivb) [inline rust/src/rfb/rfb.rs:283]: nit: improper indent

File: rust/src/rfb/rfb.rs [hunk 6]
```diff
@@ -388,7 +430,11 @@ impl RFBState {
                             consumed += current.len() - rem.len();
                             current = rem;
 
-                            self.state = parser::RFBGlobalState::TSVncResponse;
+                        if let Some(frame) = &_pdu {
+                            frame.set_len(flow, consumed as i64);
+                        }
+
+                        self.state = parser::RFBGlobalState::TSVncResponse;
```
Comment (inashivb) [inline rust/src/rfb/rfb.rs:463]: Same comments for this fn as above.

**Review by inashivb — CHANGES_REQUESTED** [review]

File: rust/src/rfb/rfb.rs (hunk 4 above)
Comment (hsadia538) [reply rust/src/rfb/rfb.rs:256]: I was under the impression that it will be fixed after rustfmt

**Pull request closed without merging**